addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,240 @@
+        string='Nilvera Document UUID',
+        copy=False,
+        readonly=True,
+        default=lambda self: str(uuid.uuid4()),
+        help="Universally unique identifier of the Invoice",
+    )
+    l10n_tr_nilvera_send_status = fields.Selection(
+        selection=[
+            ('error', "Error (check chatter)"),
+            ('not_sent', "Not sent"),
+            ('sent', "Sent and waiting response"),
+            ('succeed', "Successful"),
+            ('waiting', "Waiting"),
+            ('unknown', "Unknown"),
+        ],
+        string="Nilvera Status",
+        readonly=True,
+        copy=False,
+        default='not_sent',
+    )
+
+    @api.model
+    def _get_ubl_cii_builder_from_xml_tree(self, tree):
+        customization_id = tree.find('{*}CustomizationID')
+        if customization_id is not None and 'TR1.2' in customization_id.text:
+            return self.env['account.edi.xml.ubl.tr']
+        return super()._get_ubl_cii_builder_from_xml_tree(tree)
+
+    def button_draft(self):
+        # EXTENDS account
+        for move in self:
+            if move.l10n_tr_nilvera_uuid and move.l10n_tr_nilvera_send_status != 'not_sent':
+                raise UserError(_("You cannot reset to draft an entry that has been sent to Nilvera."))
+        super().button_draft()
+
+    def _l10n_tr_nilvera_submit_einvoice(self, xml_file, customer_alias):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint=f"/einvoice/Send/Xml?Alias={customer_alias}",
+            post_series_func=self._l10n_tr_nilvera_post_einvoice_series,
+        )
+
+    def _l10n_tr_nilvera_submit_earchive(self, xml_file):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint="/earchive/Send/Xml",
+            post_series_func=self._l10n_tr_nilvera_post_earchive_series,
+        )
+
+    def _l10n_tr_nilvera_submit_document(self, xml_file, endpoint, post_series_func=None):
+        """Generic function to handle e-invoice and e-archive submissions to Nilvera."""
+        client = _get_nilvera_client(self.env.company)
+        response = client.request(
+            "POST",
+            endpoint,
+            files={'file': (xml_file.name, xml_file, 'application/xml')},
+            handle_response=False,
+        )
+
+        if response.status_code == 200:
+            self.l10n_tr_nilvera_send_status = 'sent'
+        elif response.status_code in [401, 403]:
+            raise UserError(_("Oops, seems like you're unauthorised to do this. Try another API key with more rights or contact Nilvera."))
+        elif 400 <= response.status_code < 500:
+            error_message, error_codes = self._l10n_tr_nilvera_einvoice_get_error_messages_from_response(response)
+
+            # If the sequence/series is not found on Nilvera, add it then retry.
+            if 3009 in error_codes and post_series_func:
+                post_series_func(client)
```

Comment:
Last security concern here: users can call `_l10n_tr_nilvera_submit_document` with an arbitrary function as parameter and it might execute meaning they can retrieve the session to use it at will and obtain data.